exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyGroupScan.java

```diff
@@ -172,6 +175,26 @@ private EasyGroupScan(final EasyGroupScan that) {
     mappings = that.mappings;
     partitionDepth = that.partitionDepth;
     metadataProvider = that.metadataProvider;
+    maxRecords = that.maxRecords;
+    supportsLimitPushdown = that.formatPlugin.easyConfig().supportsLimitPushdown;
+  }
+
+  // Constructor to get the limit pushed down
+  private EasyGroupScan(final EasyGroupScan that, int maxRecords) {
```

Comment:
Please do not introduce the new constructor. `maxRecords` is not final, so it is possible to assign its value after using the existing copy constructor.

Reply:
Fixed